diff --git a/cpp/src/quantity.cpp b/cpp/src/quantity.cpp
index 352fad0..0a469de 100644
--- a/cpp/src/quantity.cpp
+++ b/cpp/src/quantity.cpp
@@ -1,6 +1,7 @@
 #include "quantity.h"
 
 #include <sstream>
+#include <algorithm>
 
 namespace fnss {
 
@@ -20,11 +21,13 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
 	value(0), unit(converter_.getBaseUnit()), converter(converter_) {}
 
 void Quantity::fromString(const std::string &str) {
-	std::istringstream ss(str);
-	ss>>this->value>>this->unit;
+        std::string::size_type sz;
+        this->value = std::stod (str,&sz);
+        this->unit = str.substr(sz);
+        this->unit.erase(remove_if(this->unit.begin(), this->unit.end(), isspace), this->unit.end());
 
-	if(this->unit == "")
-		this->unit = this->converter.getBaseUnit();
+        if(this->unit == "")
+                this->unit = this->converter.getBaseUnit();
 }
 
 std::string Quantity::toString(const std::string &separator) const {
