diff --git a/Makefile b/Makefile
index 1f6582c..202380f 100644
--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,8 @@ all : libhiberlite.a sqlite3.o tests sample
 
 OBJS=BeanLoader.o BeanUpdater.o ChildKiller.o CppModel.o Database.o ModelExtractor.o Registry.o SQLiteStmt.o Visitor.o shared_res.o sqlite3.o
 
-CXXFLAGS=-std=c++0x -Iinclude/ -Wall -Isqlite-amalgamation
-LDFLAGS=-lpthread -ldl
+CXXFLAGS=-std=c++0x -fPIC -Iinclude/ -Wall -Isqlite-amalgamation
+LDFLAGS=-lpthread -ldl -fPIC
 
 libhiberlite.a : $(OBJS)
 	ar -r -s libhiberlite.a $(OBJS)
@@ -22,7 +22,7 @@ install :
 	cp libhiberlite.a $(INSTALL_LIB)/
 
 sqlite3.o :
-	gcc -c sqlite-amalgamation/sqlite3.c -o sqlite3.o
+	gcc -c -fPIC sqlite-amalgamation/sqlite3.c -o sqlite3.o
 
 %.o : src/%.cpp include/*
 	g++ -c $(CXXFLAGS) $< -o $@
diff --git a/include/hiberdefs.h b/include/hiberdefs.h
index 1293ce9..45f22dc 100644
--- a/include/hiberdefs.h
+++ b/include/hiberdefs.h
@@ -8,13 +8,20 @@
 #define HIBERLITE_NVP(Field) hiberlite::sql_nvp< decltype(Field) >(#Field,Field)
 #endif
 #define HIBERLITE_BASE_CLASS(ClName) hiberlite::sql_nvp< ClName >(#ClName,*((ClName*)this) )
+#define HIBERLITE_BASE_CLASS_WITH_NAME(ClName,tName) hiberlite::sql_nvp< ClName >(#tName,*((ClName*)this))
 
-#define HIBERLITE_EXPORT_CLASS(ClName) 			\
+#define HIBERLITE_EXPORT_CLASS(ClName)      \
 namespace hiberlite{							\
 template<>										\
 std::string Database::getClassName<ClName>()	\
 {	std::string temp(#ClName);std::replace(temp.begin(), temp.end(), ':', '_');return temp;}}
 
+#define HIBERLITE_EXPORT_CLASS_WITH_NAME(ClName,tName)       \
+namespace hiberlite{             \
+template<>                   \
+std::string Database::getClassName<ClName>() \
+{  return #tName;  }}
+
 //#define HIBERLITE_COLLECTION(Field) hiberlite::collection_nvp<typeof(Field),typeof(Field.begin())>(#Field, Field, Field.begin(), Field.end())
 //#define HIBERLITE_ARRAY(Field,N) hiberlite::collection_nvp<typeof(Field),typeof(Field[0])>(#Field,  *Field, *(Field+N))
 
diff --git a/include/some_types.h b/include/some_types.h
index e5dbc70..f938f31 100644
--- a/include/some_types.h
+++ b/include/some_types.h
@@ -1,6 +1,8 @@
 #ifndef SOM_TYPES_H_INCLUDED
 #define SOM_TYPES_H_INCLUDED

+#include <cstdint>
+
 namespace hiberlite{

 template<class E, class C>
